clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnectionBuilder.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+package com.microsoft.aspnet.signalr;
+
+public class HubConnectionBuilder {
+    private boolean built = false;
+    private String url;
+    private Transport transport;
+    private Logger logger;
+
+    public HubConnectionBuilder setUrl(String url){
```

Comment:
We should match the .NET API. So `withUrl`. Also, can we put the transport as a parameter here so that we're matching the behavior of the C# client?

Reply:
I personally hate the `withUrl` api we have in the C# client 😄

Reply:
I'm in favor of using excplicit calls to `withUrl`, and `withTransport`

Reply:
How will that work with things like raw tcp transports? The idea of the `WithUrl` call in C# is that that is the call that sets up the "HTTP Transport" layer, which needs a URL and (optionally) the kind of Http Transport to use.

In C#, the TCP code would be something like:

```
var connection = new HubConnectionBuilder()
    .WithEndpoint("example.com", port: 1234)
    .Build()
```

So there wouldn't be an option to declare a transport. If we have `withTransport` separately, then you could get in to trouble with things like:

```java
HubConnection connection = new HubConnectionBuilder()
    .withEndpoint("example.com", port: 1234)
    .withTransport(LongPolling) // What does this line mean now?
    .Build()
```

Reply:
*sigh Okay

Reply:
I still don't like `withUrl` though lol

Reply:
I don't entirely disagree with that :).